### ISSUE TITLE
Next headers cookies has is not defined after the set cookies in app router

### ISSUE THREAD

**Issue by VitaliyR** [vercel/next.js#53322]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:20 PDT 2023; root:xnu-8796.121.3~7/RELEASE_ARM64_T6000
    Binaries:
      Node: 18.16.1
      npm: 9.5.1
      Yarn: N/A
      pnpm: N/A
    Relevant Packages:
      next: 13.4.13-canary.6
      eslint-config-next: 13.4.12
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: standalone
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App Router, Routing (next/router, next/navigation, next/link)

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/VitaliyR/nextjs-13-cookies-has-issue/blob/main/app/page.tsx

### To Reproduce

Open the provided link and see it in action.

Reproduction scenario:
App Routes, Server Components.

Cookies here - `import { cookies } from 'next/navigation'`

1. In your server component use `cookies().has` method - on original request works fine
2. Alter your cookie value (e.g. call `cookies().set` or `cookies().delete`)
3. Crashes with error that `not a function` because `has` is undefined in the result of `cookies()` call

### Describe the Bug

It seems that in server components after cookies has been altered - result of `cookies()` object is different compared to the result from the original call.

So basically, you have a component which console logs `typeof cookies().has` and a button which alters cookies in server action (set or delete).

On original call (no action, just landing on the page) the console output will be predictable - `function`. So, `cookies().has` is defined.

After user clicks on the button which runs server action and alters the cookies - crash happens, and console output is `undefined` - meaning that there are no `has` property in `cookies()` result object.

`cookies().get` is present in both cases - original land and cookies update.

### Expected Behavior

`cookies().has` should be always defined as a function and works as expected

**Comment by shrekuu** [comment]
Note this still exists in v13.4.13.  For the moment, try `const secret = cookies().get('secret_key')` then try to check `secret`.

**Comment by VitaliyR** [comment]
Sorry for being late on the party.
I can confirm that this has been fixed. I've tested in `15.1.7` - not reproducing anymore.
Thanks!
